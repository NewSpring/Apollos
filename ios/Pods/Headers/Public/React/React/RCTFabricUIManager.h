../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h