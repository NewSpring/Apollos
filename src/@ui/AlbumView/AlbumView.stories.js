import React from 'react';
import { storiesOf } from '@storybook/react-native';

import AlbumView from './';

storiesOf('AlbumView', module)
  .add('basic', () => (
    <AlbumView
      title="Able"
      artist="NewSpring"
      blurredImage={{
        fileLabel: null,
        fileName: 'able.blurred.1x1.jpg',
        fileType: null,
        url: '//drhztd8q3iayu.cloudfront.net/newspring/collection/albums/able.blurred.1x1.large.jpg',
      }}
      albumImage={{
        fileLabel: null,
        fileName: 'able.artwork.1x1.jpg',
        fileType: null,
        url: '//drhztd8q3iayu.cloudfront.net/newspring/collection/albums/able.artwork.1x1.large.jpg',
      }}
    />
  ));
