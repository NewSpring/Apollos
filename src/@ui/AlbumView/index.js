import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import { View, Platform, StyleSheet } from 'react-native';
import PaddedView from '@ui/PaddedView';
import ProgressiveImage from '@ui/ProgressiveImage';
import { H5, H7 } from '@ui/typography';
import { withThemeMixin } from '@ui/theme';
import styled from '@ui/styled';

const Container = styled(({ theme }) => ({
  backgroundColor: theme.colors.background.default,
}), 'AlbumView.Container')(View);

const InfoWrapper = styled({
  flexDirection: 'row', alignItems: 'stretch',
}, 'AlbumView.InfoWrapper')(PaddedView);

const AlbumArt = styled({
  position: 'relative',
  width: '33%',
  aspectRatio: 1,
  ...Platform.select({
    web: {
      height: 0,
      paddingTop: '100%',
    },
  }),
}, 'AlbumView.AlbumArt')(ProgressiveImage);

const BlurredImage = styled(({ theme }) => ({
  ...StyleSheet.absoluteFillObject,
  opacity: theme.alpha.low,
}), 'AlbumView.BlurredImage')(ProgressiveImage);

const TitleWrapper = styled({
  width: '66%',
  justifyContent: 'space-around',
}, 'AlbumView.TitleWrapper')(PaddedView);

const Title = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Title')(H5);

const Artist = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Artist')(H7);

const enhance = compose(
  withThemeMixin({ type: 'dark' }),
  pure,
  setPropTypes({
    title: PropTypes.string,
    artist: PropTypes.string,
    blurredImage: ProgressiveImage.propTypes.source,
    albumImage: ProgressiveImage.propTypes.source,
    isLoading: PropTypes.bool,
  }),
);

const AlbumView = enhance(({
  title, artist, blurredImage, albumImage, isLoading = false,
}) => (
  <Container>
    {(blurredImage) ? <BlurredImage source={blurredImage} /> : null}
    <InfoWrapper>
      <AlbumArt isLoading={isLoading} source={albumImage} />
      <TitleWrapper>
        <Title>{title}</Title>
        <Artist>{artist}</Artist>
      </TitleWrapper>
    </InfoWrapper>
  </Container>
));

export default AlbumView;
