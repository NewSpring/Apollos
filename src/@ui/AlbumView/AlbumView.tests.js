import React from 'react';
import renderer from 'react-test-renderer';
import Providers from '@ui/TestProviders';

import AlbumView from './';

describe('The AlbumView component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <AlbumView
          title="Able"
          artist="NewSpring"
          blurredImage={{
            fileLabel: null,
            fileName: 'able.blurred.1x1.jpg',
            fileType: null,
            url: '//drhztd8q3iayu.cloudfront.net/newspring/collection/albums/able.blurred.1x1.large.jpg',
          }}
          albumImage={{
            fileLabel: null,
            fileName: 'able.artwork.1x1.jpg',
            fileType: null,
            url: '//drhztd8q3iayu.cloudfront.net/newspring/collection/albums/able.artwork.1x1.large.jpg',
          }}
        />
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
