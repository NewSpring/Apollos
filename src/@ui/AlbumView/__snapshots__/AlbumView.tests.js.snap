// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AlbumView component should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#303030",
    }
  }
>
  <View
    style={
      Object {
        "aspectRatio": 1,
        "backgroundColor": "#858585",
        "bottom": 0,
        "left": 0,
        "opacity": 0.4,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#858585",
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    <View
      style={
        Object {
          "aspectRatio": 1,
          "backgroundColor": "#858585",
          "position": "relative",
          "width": "33%",
        }
      }
    >
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "aspectRatio": 1,
              "backgroundColor": "#858585",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "justifyContent": "space-around",
          "paddingHorizontal": 20,
          "paddingVertical": 20,
          "width": "66%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#ffffff",
            "flexWrap": "wrap",
            "fontFamily": "Colfax-Bold",
            "fontSize": 20.02,
            "lineHeight": 25.33,
            "paddingBottom": 3.15,
            "paddingTop": 4.5,
          }
        }
      >
        Able
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#ffffff",
            "flexWrap": "wrap",
            "fontFamily": "Colfax-Regular",
            "fontSize": 14,
            "lineHeight": 15.78,
            "paddingBottom": 1.8,
            "paddingTop": 3.6,
          }
        }
      >
        NewSpring
      </Text>
    </View>
  </View>
</View>
`;
